torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -1071,17 +1071,29 @@ class ShapePropagator {
 
     static const auto factory_with_ndim = [](Node* node,
                                              int dim) -> type_vec_t {
-      auto maybe_layout = node->get<at::Layout>(attr::layout);
-      if (!maybe_layout || maybe_layout != at::kStrided)
+      at::optional<IValue> maybe_layout_option = node->get(attr::layout);
+      if (!maybe_layout_option)
```

Comment:
why is there different behavior for all of these if the optional is false vs if the Node is None?

Reply:
The way I understand this get returns a nullptr if the attribute isn't defined (i.e. the argument isn't set) which is then stored in maybe_layout_option. If the argument does exist, we set a default if it's set to None. This applies to all three arguments.

cc @zdevito